app/models/concerns/second_factor_manager.rb

```diff
@@ -68,26 +93,98 @@ def security_keys_enabled?
   end
 
   def has_multiple_second_factor_methods?
-    security_keys_enabled? && (totp_enabled? || backup_codes_enabled?)
+    security_keys_enabled? && totp_or_backup_codes_enabled?
+  end
+
+  def totp_or_backup_codes_enabled?
+    totp_enabled? || backup_codes_enabled?
+  end
+
+  def only_security_keys_enabled?
+    security_keys_enabled? && !totp_or_backup_codes_enabled?
+  end
+
+  def only_totp_or_backup_codes_enabled?
+    !security_keys_enabled? && totp_or_backup_codes_enabled?
   end
 
   def remaining_backup_codes
     self&.user_second_factors&.backup_codes&.count
   end
 
   def authenticate_second_factor(token, second_factor_method)
+    second_factor_method = second_factor_method.to_i if !second_factor_method.is_a?(Integer)
```

Comment:
Any reasons we need that test?

```suggestion
    second_factor_method = second_factor_method.to_i
```